snuba/manual_jobs/scrub_ips_from_spans.py

```diff
@@ -0,0 +1,49 @@
+from datetime import datetime
+from typing import Any, Mapping, Optional
+
+from snuba.clusters.cluster import ClickhouseClientSettings, get_cluster
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.manual_jobs import Job, JobLogger, JobSpec
+
+
+class ScrubIpFromSentryTags(Job):
+    def __init__(self, job_spec: JobSpec) -> None:
+        self.__validate_job_params(job_spec.params)
+        super().__init__(job_spec)
+
+    def __validate_job_params(self, params: Optional[Mapping[Any, Any]]) -> None:
+        assert params
+        assert isinstance(params["project_ids"], list)
+        assert [isinstance(p, (int, str)) for p in params["project_ids"]]
+        self._project_ids = params["project_ids"]
+        self._start_datetime = datetime.fromisoformat(params["start_datetime"])
+        self._end_datetime = datetime.fromisoformat(params["end_datetime"])
+
+    def _get_query(self, cluster_name: str | None) -> str:
+        project_ids = ",".join([str(p) for p in self._project_ids])
+        start_datetime = self._start_datetime.isoformat()
+        end_datetime = self._end_datetime.isoformat()
+        on_cluster = f"ON CLUSTER '{cluster_name}'" if cluster_name else ""
+        return f"""ALTER TABLE spans_local
+{on_cluster}
+UPDATE `sentry_tags.value` = arrayMap((k, v) -> if(k = 'user.ip', 'scrubbed', v), `sentry_tags.key`, `sentry_tags.value`)
+WHERE project_id IN [{project_ids}]
+AND has(`sentry_tags.key`, 'user.ip')
+AND end_timestamp > toDateTime('{start_datetime}')
+AND end_timestamp <= toDateTime('{end_datetime}')"""
```

Comment:
I think `connection.execute` will do some interpolation for you, though I don't know if it's sophisticated enough to translate the project_ids array into a ClickHouse SQL array: 

See https://github.com/getsentry/snuba/blob/17a19ad9d020d25f847dd3fabd94f8383519a666/snuba/manual_jobs/update_migration_status.py#L78-L91 for an example.

Not a deal-killer, but I think it would be nice if we could get the custom interpolation out of here.

Reply:
on the subject of future improvements: this could be more general by making `'user.ip'` a parameter, and then the job is "scrub arbitrary field from spans" (which could be further generalized if we took the cluster as a parameter).

Reply:
Yeah I thought about generalizing but It's not especially complicated to do later